src/test/javascript/spec/component/complaints/complaint-student-view.component.spec.ts

```diff
@@ -0,0 +1,274 @@
+import { Result } from 'app/entities/result.model';
+import { Exam } from 'app/entities/exam.model';
+import { Submission } from 'app/entities/submission.model';
+import { Complaint } from 'app/entities/complaint.model';
+import { Observable, of } from 'rxjs';
+import { Course } from 'app/entities/course.model';
+import { ComplaintsStudentViewComponent } from 'app/complaints/complaints-for-students/complaints-student-view.component';
+import { ComplaintsFormComponent } from 'app/complaints/form/complaints-form.component';
+import { ComplaintRequestComponent } from 'app/complaints/request/complaint-request.component';
+import { ComplaintResponseComponent } from 'app/complaints/response/complaint-response.component';
+import { AccountService } from 'app/core/auth/account.service';
+import { User } from 'app/core/user/user.model';
+import { MockAccountService } from '../../helpers/mocks/service/mock-account.service';
+import { ArtemisServerDateService } from 'app/shared/server-date.service';
+import dayjs from 'dayjs';
+
+describe('ComplaintInteractionsComponent', () => {
+    const complaintTimeLimitDays = 7;
+    const course: Course = {
+        id: 1,
+        complaintsEnabled: true,
+        maxComplaintTimeDays: complaintTimeLimitDays,
+        requestMoreFeedbackEnabled: true,
+        maxRequestMoreFeedbackTimeDays: complaintTimeLimitDays,
+    };
+    const courseWithoutFeedback: Course = { id: 1, complaintsEnabled: true, maxComplaintTimeDays: 7, requestMoreFeedbackEnabled: false };
+    const examExercise: Exercise = { id: 1, teamMode: false, course } as Exercise;
+    const courseExercise: Exercise = { id: 1, teamMode: false, course, assessmentDueDate: dayjs().subtract(1, 'day') } as Exercise;
+    const submission: Submission = {} as Submission;
+    const result: Result = { id: 1, completionDate: dayjs().subtract(complaintTimeLimitDays - 1, 'day') } as Result;
+    const resultWithoutCompletionDate: Result = { id: 1 } as Result;
+    const user: User = { id: 1337 } as User;
+    const participation: Participation = { id: 2, results: [result], submissions: [submission], student: user } as Participation;
+    const defaultExam: Exam = {
+        examStudentReviewStart: dayjs().subtract(complaintTimeLimitDays, 'day'),
+        examStudentReviewEnd: dayjs().add(complaintTimeLimitDays, 'day'),
+    } as Exam;
+    const complaint = new Complaint();
+    const numberOfComplaints = 42;
+
+    let component: ComplaintsStudentViewComponent;
+    let fixture: ComponentFixture<ComplaintsStudentViewComponent>;
+    let complaintService: ComplaintService;
+    let accountService: AccountService;
+    let serverDateService: ArtemisServerDateService;
+    let numberOfAllowedComplaintsStub: jest.SpyInstance<Observable<number>, [courseId: number, teamMode?: boolean | undefined]>;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [TranslateModule.forRoot(), ArtemisTestModule],
```

Comment:
is `TranslateModule.forRoot()` really needed?

Reply:
No, removed it.